### DISCUSSION TITLE
Error starting service.

### DISCUSSION THREAD

**Discussion by zhubao315** [open-webui/open-webui#918] [General]
OSError: We couldn't connect to 'https://huggingface.co' to load this file, couldn't find it in the cached files and it looks like sentence-transformers/all-MiniLM-L6-v2 is not the path to a directory containing a file named config.json.
Checkout your internet connection or see how to run the library in offline mode at 'https://huggingface.co/docs/transformers/installation#offline-mode'.
No WEBUI_SECRET_KEY provided
Loading WEBUI_SECRET_KEY from .webui_secret_key
No sentence-transformers model found with name sentence-transformers/all-MiniLM-L6-v2. Creating a new one with MEAN pooling.

**Comment by small-ora** [comment]
I also encountered this problem. Have you found a solution?

**Reply by fewhub** [reply]
I guess you mounted the volume during docker startup, don't mount the volume for the first startup because the mounted empty directory will overwrite the files in the container, after normal startup, execute the command: docker cp open-webui:/app/backend/data .  Copy the data folder out, and then start docker again, this time you can mount the data into the container

**Reply by small-ora** [reply]
It's worked!Thanks for your help very mush!

**Reply by Eforen** [reply]
This is extreamly inconvenient for anyone using something like K8s.

**Reply by justinh-rahb** [reply]
Related:
- #1076

**Reply by Eforen** [reply]
I just confirmed it is not related

**Reply by Eforen** [reply]
I thought so too initially though.

**Reply by Eforen** [reply]
If this is a known behavior then why is it not in the readme how to handle this correctly? It seems to me that if someone used the instructions in the readme this would still be an issue for them...

**Reply by justinh-rahb** [reply]
Because this is a volunteer developed project, worked on in people's spare time, for no salary. What are you expecting? If you've got a better idea of how to document/configure this... by all means, open a PR. Unsubscribing this thread.

**Reply by Eforen** [reply]
I did

**Reply by liyimeng** [reply]
this is coz the program wanna read /app/backend/data/cached dir, why you are likely mount a volume into /app/backend/data and overwrite its content when running with container.

**Comment by Saigut** [comment]
How is this issue going?

**Reply by hakanisaksson** [reply]
Hi, I got the same error when trying to run the container in k8s (in an air gapped cluster); using latest main image from 2days ago.

`OSError: We couldn't connect to 'https://huggingface.co' to load this file, couldn't find it in the cached files and it looks like sentence-transformers/all-MiniLM-L6-v2 is not the path to a directory containing a file named config.json.
`

First tried the fix mentioned in https://github.com/open-webui/open-webui/issues/1076
That did help, but did not fix the issue.
Then added this enviroment variable after https://github.com/open-webui/open-webui/blob/main/kubernetes/helm/templates/webui-deployment.yaml#L44:
```
        - name: RAG_EMBEDDING_MODEL
          value: "/app/backend/data/cache/embedding/models/models--sentence-transformers--all-MiniLM-L6-v2/snapshots/e4ce9877abf3edfe10b0d82785e83bdcb973e22e"

```
That solved the issue with running open-webui in k8s for me at least, until there is a better fix.

**Reply by ghost** [reply]
Take a look at #1419, it aims to eliminate these connections on startup. One caveat to hard-coding `RAG_EMBEDDING_MODEL` to use the filesystem snapshot path is that it will change depending on which revision is selected when the model was loaded and container was built. My PR attempts to solve it with a new dedicated function which accepts the model shortname.

You may also be interested in #1302 and my recent [comment](https://github.com/open-webui/open-webui/issues/1302#issuecomment-2039091310) regarding `LITELLM_LOCAL_MODEL_COST_MAP`:

> I've confirmed that `--env LITELLM_LOCAL_MODEL_COST_MAP="True"` successfully eliminates the LiteLLM startup connection. When used along with #1419, there are no connections on startup except for your configured LLM models (in my case localhost for Ollama). The Open-WebUI update check still must be manually triggered via the Settings -> About page. Offline startup us MUCH faster!

**Reply by Saigut** [reply]
@lainedfles Why does it need to download a model? Shouldn't it be up to the user to decide whether to download?

**Reply by ghost** [reply]
@Saigut Hugging Face model download is required for the Document RAG and Whisper TTS features. However, the pre-built official Docker images pre-load the models by downloading them during the build process. As long as model auto-update is disabled, (the default of `RAG_EMBEDDING_MODEL_AUTO_UPDATE` is false), download is only required when manually updating the model within the GUI setting or when building a Docker image yourself or installing manually.

**Comment by mamawr** [comment]
The enviroment variable RAG_EMBEDDING_MODEL_AUTO_UPDATE=True can help to fix this issue

**Reply by eric1lubow** [reply]
I fixed it  by add arg :  
 **-e HF_ENDPOINT=https://hf-mirror.com**

It's my full cmd:
#docker run -d -e HF_ENDPOINT=https://hf-mirror.com -p 3000:8080 -e OLLAMA_BASE_URL=http://192.168.2.26:11434 -v open-webui/backend/data:/app/backend/data --name open-webui  ghcr.io/open-webui/open-webui:main

**Reply by ghost** [reply]
@mamawr while it can help to download a model automatically on startup, understand that leaving this permanently enabled may invalidate your ChromaDB vector collection if/when the model is automatically updated. In this event, the vector database must be cleared and all documents must be re-added or re-scanned.

**Comment by ZiHAO-LI-cmd** [comment]
It's a network problem that you cannot download model from Huggingface.
You can try to add `HF_ENDPOINT=https://hf-mirror.com` in `.env` file, so that model will be downloaded from the mirror site.

**Reply by Saigut** [reply]
Why does it need to download a model? Shouldn't it be up to the user to decide whether to download?

**Reply by ZiHAO-LI-cmd** [reply]
Open WebUI supports uploading documents for RAG function, which needs an embedding model. The downloaded `all-MinLM-L6-v2` is actually an embedding model for vectorizing splited file text.

**Comment by glx2013** [comment]
https://www.cnblogs.com/dajianshi/p/18118969#:~:text=%E4%B8%80%E4%B8%AA%E5%91%BD%E4%BB%A4%E8%A1%8C%E5%8F%82%E6%95%B0%E8%A7%A3%E5%86%B3o

export HF_ENDPOINT=https://hf-mirror.com
